website/source/docs/concepts/tokens.html.md

```diff
@@ -42,14 +42,53 @@ Finally, the `-policy` flag can be used to set the policies associated
 with the token. Learn more about policies on the
 [policies concepts](/docs/concepts/policies.html) page.
 
+<a name="ttls-and-leases" />
 ## Token Time-To-Live and Leases
 
-Every non-root token has a time-to-live (TTL) associated with it.  After the
-TTL is up, the token will no longer function. In addition to no longer
-functioning, Vault will revoke it.
+Every non-root token has a time-to-live (TTL) associated with it, which is
+the number of seconds the token is valid for, _measured from the moment it is
+created_. After the TTL is up, the token will no longer function, and
```

Comment:
The field was changed to TTL from Lease, but as you can see the internal JSON tag `lease` has stuck around because at least right now Go won't let you define the same tag for two members. But that's really all internal so not a big problem. The only reason it really changed in the code was to make it more understandable for other people looking at the code (or the Godoc representation of the package) exactly what that value was, since having `Lease` in `LeaseOptions` was confusing.

The message about lease vs. TTL is actually decently simple: just like if you lease a car, or lease an apartment, a lease is granting you the right to use something -- in this case credentials (whether a Vault token itself, or dynamic credentials generated by the Vault token) -- for some period of time, which used to also be called `lease` but is now referred to as the TTL because it was just confusing. That period of time is also called the `lease_duration` in JSON for backwards compatibility, but we didn't feel strongly about changing that as it's also a valid description of the concept.

When you lease an apartment, chances are your landlord will let you stay on for another year, if you've not trashed the place. So that's renewing. Generally this will be for a period of time equivalent to the default TTL, but you can request a different period of time -- and just like the landlord, Vault may or may not honor that request, so you need to actually check. If the landlord said "nope, only 6 months instead of 12" and you ignored her, you can't be annoyed when you get the eviction notice.

After some years the landlord might say "sorry -- actually, after 6 more months I need the apartment empty to renovate the whole thing, so you gotta go". That's your max TTL. And just like in real life, that max TTL can change...for instance, let's say the landlord couldn't get the contractors to start on time so you get a six month reprieve. But you don't know for sure ahead of time. In the case of Vault, the administrator might in the interim say "oh, actually, we want secrets issued from mount/role X to be valid for a max of A instead of B". Maybe that's smaller, maybe that's larger, but in nearly all cases Vault reevaluates the max TTL based on the _current_ settings and checks that against the creation time. (In a near release there may be one exception to this coming in token roles where a token can have a definite, explicit max that never changes.)